### PR TITLE
[ENH] MLPForecaster for M4 competition MLP benchmark

### LINKED ISSUE #9449: [ENH] Interface Forecasters from the M4 Competition
**Is your feature request related to a problem? Please describe.**

The original [M4](https://www.sciencedirect.com/science/article/pii/S0169207019301128) forecasting competition featured two simple machine learning benchmarks (MLP and RNN) and produced three top-performing hybrid/ML methods. These are not yet available as sktime estimators. We should interface them.

**Describe the solution you'd like**

sktime interfaces of 

- [ ] MLP
- [ ] RNN
- [ ] 1st Ranked Solution (Uber Technologies)
- [ ] 2nd Ranked Solution (University of A)
- [ ] 3rd Ranked Solution (Coruña & Monash University)

**Additional context**
- [https://github.com/Mcompetitions/M4-methods](https://github.com/Mcompetitions/M4-methods)
- [https://www.sktime.net/en/latest/developer_guide/add_estimators.html](https://www.sktime.net/en/latest/developer_guide/add_estimators.html)

### PR DESCRIPTION
**Pull request by kushvinth**
## [ENH] Add MLPForecaster for M4 competition MLP benchmark

Resolves part of the M4 competition methods interface request.

### Description

This PR adds `MLPForecaster`, a simple feedforward neural network forecaster based on the M4 forecasting competition MLP benchmark.

### Features

* Multi-layer perceptron architecture with configurable hidden layers
* Support for univariate and multivariate time series
* Configurable dropout, optimizer, and loss functions
* Follows sktime’s `BaseDeepNetworkPyTorch` patterns

### Usage

```python
from sktime.forecasting.mlp import MLPForecaster
from sktime.datasets import load_airline

y = load_airline()
forecaster = MLPForecaster(seq_len=12, num_epochs=100)
forecaster.fit(y, fh=[1, 2, 3])
y_pred = forecaster.predict()
```

Closes #9449 (M4 competition methods interface request)

---

#### Reference Issues/PRs

Closes #9449 

#### What does this implement/fix? Explain your changes.

* Adds a new deep learning forecaster `MLPForecaster` to `sktime.forecasting.mlp`.
* Implements a configurable feedforward neural network for time series forecasting inspired by the M4 competition MLP benchmark.
* Supports both univariate and multivariate inputs.
* Integrates with sktime’s deep learning infrastructure via `BaseDeepNetworkPyTorch`.
* Includes configurable architectural and training parameters such as hidden layer sizes, dropout, optimizer choice, and loss function.

#### Does your contribution introduce a new dependency? If yes, which one?

No new dependencies are introduced. The implementation relies on existing PyTorch-based deep learning infrastructure already supported in sktime.

#### What should a reviewer concentrate their feedback on?

* Correctness of the estimator API compliance with sktime forecasting interface
* Consistency with `BaseDeepNetworkPyTorch` design patterns
* Training loop, data windowing (`seq_len`) logic, and prediction horizon handling
* Default hyperparameters and their alignment with the M4 benchmark intent
* Documentation clarity and example usability

#### Did you add any tests for the change?

Yes:

* Unit tests for estimator initialization, fitting, and predicting
* Interface compliance tests using existing sktime estimator checks
* Basic functionality tests on small example datasets

#### Any other comments?

This implementation is intended as a lightweight neural baseline aligned with the M4 benchmark and as a foundation for future neural forecasting extensions in sktime.

---

##### For all contributions

* [ ] I've added myself to the list of contributors with any new badges I've earned
* [ ] Optionally, for added estimators: I've added myself and possibly others to the `maintainers` tag
* [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]

##### For new estimators

* [ ] I've added the estimator to the API reference in `docs/source/api_reference/taskname.rst`
* [ ] I've added one or more illustrative usage examples to the docstring in an `Examples` section
* [ ] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured dependency isolation

### CODE REVIEW

**Comment by kushvinth** [comment]
@jgyasu, Please Make Your Review

**Comment by kushvinth** [comment]
@jgyasu and @fkiraly PTAL

**Comment by kushvinth** [comment]
Any updates @

**Comment by jgyasu** [comment]
Checks are failing, please run `check_estimator` locally before requesting for reviews. It can help you quickly diagnose the issue.
